store-api-server/src/payment/controller/payment.controller.ts

```diff
@@ -124,60 +125,110 @@ export class PaymentController {
+   * @apiSuccessExample Example Success-Response:
+   *  {
+   *    "tezpay": {
+   *        "id": "pi_3L2foUEdRDZNp7JF0yajTKrM",
+   *        "amount": "0.66",
+   *        "currency": "XTZ",
+   *        "vatRate": 0.1,
+   *        "amountExclVat": "0.55",
+   *        "providerPaymentDetails": {
+   *          "paypointMessage": "pi_3L2foUEdRDZNp7JF0yajTKrM_secret_cs3SWGJy7xdd5M19EwShCjbdE",
+   *          "receiverAddress": "KT1..",
+   *          "mutezAmount": "660000"
+   *        }
+   *    },
+   *    "stripe": {
+   *        ...
+   *    }
+   *  }
+   *  @apiExample {http} Example http request url (make sure to replace $base_url with the store-api-server endpoint):
+   *  $base_url/payment/order-now/10
+   * @apiName orderNow
+   */
+  @Post('/order-now/:id')
+  @UseGuards(JwtAuthGuard)
+  orderNow(
+    @Session() cookieSession: any,
+    @Req() request: any,
+    @CurrentUser() user: UserEntity,
+    @Param('nftId') nftId: number,
+    @Body('paymentProviders') paymentProviders: PaymentProvider[],
+    @Body('recreateNftOrder') recreateNftOrder: boolean = false,
+    @Body('currency') currency: string = BASE_CURRENCY,
+  ) {
+    const [firstProvider, secondProvider] = paymentProviders.map((provider) => {
       return {
-        ...paymentIntent,
-        amountExclVat: this.paymentService.currencyService.toFixedDecimals(
-          paymentIntent.currency,
-          amountExclVat,
-        ),
-        paymentDetails: paymentIntent.providerPaymentDetails,
-        nfts: order.nfts,
-        expiresAt: order.expiresAt,
+        provider,
+        currency: this.paymentService.getCurrency(provider, currency),
       };
-    } catch (err: any) {
-      Logger.error(
-        `Err on creating nft order (userId=${user.id}), err: ${err}`,
-      );
+    });
 
-      if (err instanceof HttpException) {
-        throw err;
-      }
-      throw new HttpException(
-        'unable to place the order',
-        HttpStatus.INTERNAL_SERVER_ERROR,
+    this.userService
+      .handleCartAdd({ cookieSession, user, nftId })
+      .then(async () => {
+        const args = {
+          cookieSession,
+          user,
+          request,
+          recreateNftOrder,
+        };
+        const firstProviderIntent =
+          await this.paymentService.handleCreatePaymentIntent({
+            ...args,
+            paymentProvider: firstProvider.provider,
+            currency: firstProvider.currency,
+          });
+        const secondProviderIntent =
+          await this.paymentService.handleCreatePaymentIntent({
+            ...args,
+            paymentProvider: secondProvider.provider,
+            currency: secondProvider.currency,
+          });
+        return {
+          [firstProvider.provider]: firstProviderIntent,
+          [secondProvider.provider]: secondProviderIntent,
+        };
+      })
+      .catch((err) =>
+        Logger.error(
+          `Unable to create payment intents for payment providers, err: ${err}`,
+        ),
```

Comment:
I think here we are missing to throw  a `throw new HttpException(...`

Reply:
Thanks for the hint, added it!